pkg/controller/windowsmachineconfig/tracker.go

```diff
@@ -216,7 +309,7 @@ func (t *Tracker) createSecret(credentials *types.Credentials) (string, error) {
 
 	// Get and delete the secret if it already exists.
 	_, err := t.k8sclientset.CoreV1().Secrets(t.operatorNS).Get(credentials.GetInstanceId(), metav1.GetOptions{})
-	if err != nil && !k8serrors.IsNotFound(err) {
+	if err == nil || (err != nil && !k8serrors.IsNotFound(err)) {
```

Comment:
We should be careful here, this seems like it will pass if the operator does not have `get` permissions for secrets, and will always continue and delete the existing secret in that case

Reply:
If we don't have permission, we'd get an error and we won't delete the secret.